packages/upload/src/vaadin-upload-mixin.js

```diff
@@ -776,10 +800,19 @@ export const UploadMixin = (superClass) =>
         return;
       }
 
-      formData.append(file.formDataName, file, file.name);
+      let requestBody;
+      if (isRawUpload) {
+        // Raw binary upload - send file directly
+        requestBody = file;
```

Comment:
Should the file name be added as a header or something? It doesn't look like you can access it from the server currently.